hpat/datatypes/hpat_pandas_series_functions.py

```diff
@@ -514,3 +514,42 @@ def hpat_pandas_series_floordiv_number_impl(lhs, rhs):
         return hpat_pandas_series_floordiv_number_impl
 
     raise TypingError('{} The object must be a pandas.series or scalar. Given rhs: {}'.format(_func_name, rhs))
+
+
+@overload_method(SeriesType, 'take')
+def hpat_pandas_series_take(self, indices, axis=0, is_copy=False, **kwargs):
+    """
+    Pandas Series method :meth:`pandas.Series.take` implementation.
+    .. only:: developer
+       Test: python -m hpat.runtests hpat.tests.test_series.TestSeriesTake
+
+    Parameters
+    ----------
+    self: :obj:`pandas.Series`
+        input series
+    indices: :obj:`array-like`
+         indices to take positions
+    axis: :obj:`int`, :obj:`str`
+        *unsupported*
+    is_copy: :obj:`bool`
+        *unsupported*
+    Returns
+    -------
+    :obj:`pandas.Series`
+         returns :obj:`pandas.Series` object
+    """
+    _func_name = 'Method take().'
+
+    if not isinstance(self, SeriesType):
+        raise TypingError('{} The object must be a pandas.series. Given self: {}'.format(_func_name, self))
+
+    if self.index == types.none:
+        raise TypingError('{} The object must NOT be a None. Given self.index: {}'.format(_func_name, self.index))
+
+    if not isinstance(indices, types.List):
+        raise TypingError('{} The object must be a List. Given self: {}'.format(_func_name, self))
+
+    def hpat_pandas_series_take_impl(self, indices):
+        return pandas.Series(self._data.take(indices), index=[self._index[i] for i in indices])
```

Comment:
Why we need new list `self._index[i] for i in indices` here? Could we use `self._index` instead?

Reply:
We need to take elements from `self._index` according to `indices`. E.g.:
`data = [1.0, 13.0, 9.0, -1.0, 7.0]`
`index = [0, 1, 2, 3, 4]`
`indices = [1, 3]`

The result of `take()`:
`data = [13.0, -1.0]`
`index = [1, 3]`

I can't imagine solution w/o creation of new list.

Reply:
Could you please move this loop out of the line with `return` statement? I mean, create temporal value with meaningfull name, initialize it by this loop and put this variable into the line with return statement to `index`

Reply:
Of course!